explainer.md

```diff
@@ -0,0 +1,87 @@
+
+The ethos guiding the API design are:
+* If a visual transition can be defined by both the site author and the UA, the site author takes precendence.
+
+* If site authors can not define the transition, they should be able to detect if a visual transition was executed by the UA (assuming no security/privacy concerns).
+
+# Transition Cases
+The matrix of navigation cases which define whether a visual transition could be customized by site authors depends on the following characteristics:
+
+* Swipe (predictive) vs Atomic: An atomic navigation would be clicking a button in the browser UI or a programmatic navigation via [`history.pushState`](https://developer.mozilla.org/en-US/docs/Web/API/History/pushState). A swipe navigation would be gesture which shows the user a preview of the destination before initiating the navigation. This is also called predictive navigation since the user can choose to initiate the navigation based on the preview.
+
+* Same vs Cross-Document (Same-Origin): Whether site authors can customize a transition today and which APIs are necessary to support customization depends on whether the navigation is same-document or cross-document.
+
+* Same vs Cross-origin: Customization of visual transition when navigating between cross-origin sites should either not be supported or should be extremely limited for security/privacy concerns.
+
+The table below shows that currently only same-document atomic navigations can be customized.
+
+|   | Same-Document  |  Cross-Document Same-Origin | Cross-Origin  |
+|---|---|---|---|
+| Swipe Navigation | ❌ | ❌ | ❌ |
+| Atomic Navigation  | ✅ | ❌ | ❌ |
+
+Ideally all same-origin navigations should be customizable but the following API gaps remain to enable that:
+
+* Gesture API: Most platforms have a swipe gesture which is used by the UA to trigger a back/forward navigation. On Android/iOS its swiping from the screen edge; on Mac, its multi-finger swipe. Site authors currently can not consistently intercept these events. Customizing swipe navigations for both same-document and cross-document navigations requires this[^1].
+
+* Persisting Visual State: The transition requires showing visual state from both before/after the navigation. Site authors can do this for same-document navigations but not cross-document. [ViewTransition](https://github.com/WICG/view-transitions/blob/main/explainer.md#cross-document-same-origin-transitions) addresses this gap.
+
+The exact details for the above APIs is out of scope for this proposal. These are meant to show that there are cases which are currently not customizable by site authors but will be going forward. The API proposals here should work well if/when one of these cases becomes customizable.
+
+# Problem Statement
```

Comment:
In the absence of the gesture API, site should be able to disable default swipe for cross-document case also.